### PR TITLE
implement graceful shutdown on windows

### PR DESCRIPTION
**Pull request by mwrock**
Signed-off-by: Matt Wrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/core/src/os/process/windows.rs [hunk 1]
```diff
@@ -27,6 +28,10 @@ use super::{HabExitStatus, ExitStatusExt};
 
 const STILL_ACTIVE: u32 = 259;
 
+const PROCESS_EXITED: i32 = 0;
```
Comment (reset) [inline components/core/src/os/process/windows.rs]: These are private to the module but returned from a public function - they should be public so they can be referenced by name

Comment (reset) [inline components/core/src/os/process/windows.rs]: They also might be better represented by an enum if that fits with your API

Comment (mwrock) [inline components/core/src/os/process/windows.rs]: Yeah I thought about the enum but that did not mesh with the Linux side which uses the libc signal constants.

Thinking again about this maybe a better approach would be to define an enum in mod.rs that just signifies graceful or abrupt shutdown.

Comment (mwrock) [inline components/core/src/os/process/windows.rs]: Then it's clear enough to infer based on platform what actually happened

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/186462582).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/186462582).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**